test/src/main/java/org/apache/accumulo/test/functional/SplitIT.java

```diff
@@ -205,4 +219,79 @@ public void deleteSplit() throws Exception {
     }
   }
 
+  @Test
+  public void concurrentSplit() throws Exception {
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+
+      final String tableName = getUniqueNames(1)[0];
+
+      log.debug("Creating table {}", tableName);
+      c.tableOperations().create(tableName);
+
+      final int numRows = 100_000;
+      log.debug("Ingesting {} rows into {}", numRows, tableName);
+      VerifyParams params = new VerifyParams(getClientProps(), tableName, numRows);
+      TestIngest.ingest(c, params);
+
+      log.debug("Verifying {} rows ingested into {}", numRows, tableName);
+      VerifyIngest.verifyIngest(c, params);
+
+      log.debug("Creating futures that add random splits to the table");
+      ExecutorService es = Executors.newFixedThreadPool(10);
+      final int totalFutures = 100;
+      final int splitsPerFuture = 4;
+      final Set<Text> totalSplits = new HashSet<>();
+      List<Callable<Void>> tasks = new ArrayList<>(totalFutures);
+      for (int i = 0; i < totalFutures; i++) {
+        final Pair<Integer,Integer> splitBounds = getRandomSplitBounds(numRows);
+        final TreeSet<Text> splits = TestIngest.getSplitPoints(splitBounds.getFirst().longValue(),
```

Comment:
Looked at the [TestIngest.getSplitPoints method](https://github.com/apache/accumulo/blob/c5f4e2e25406e597f54638ddc596789c96a7501e/test/src/main/java/org/apache/accumulo/test/TestIngest.java#L193-L205) and it seems it has a bug where when `end-start < numsplits` it will set set `splitSize` to zero and infinite loop will happen.  In the previous sentence `end`, `start`, `numSplits`, and `splitSize` are variables in `TestIngest.getSplitPoints`.

Locally I tested this and it never printed anything and just chewed up 100% cpu.

```
    public static void main(String[] args) {
        var splits = TestIngest.getSplitPoints(500, 502, 4);
        System.out.println(splits);
    }
```

I think the way the random numbers are generated we could get something where `end-start` is less than 4 and then the test would hang.

Reply:
interesting. i can take a look into that

Reply:
There may be other solutions but the two i thought of are:

1. check to make sure the difference is greater than zero in the method, else throw an exception
2. if the difference is not greater than 0, return an empty TreeSet

@keith-turner, I don't see one option as being clearly the better choice than the other. Do you have an opinion or another possible solution?

Reply:
One reason I like option 2 is it test realistic behavior.  It seems plausible that a user may compute split points and sometimes their computation will yield zero splits points.

Reply:
Should be addressed by 130b4e5